deepinterpolation/generator_collection.py

```diff
@@ -1089,85 +1105,115 @@ def __getitem__(self, index):
+                    }
+
+    def _data_from_indexes(
+            self, video_index: int, img_index:int) -> Tuple[np.ndarray, np.ndarray]:
+        """ Generate X, Y dataset given a video_index and img_index
+        """
 
-        return local_lims, local_img
+        index_dict = self.frame_lookup[(video_index, img_index)]
+        input_index = index_dict['input_index']
+        output_frame = index_dict['output_frame']
 
-    def __get_norm_parameters__(self, index_frame):
-        local_lims, local_img = self.get_lims_id_sample_from_index(index_frame)
-        local_mean = self.frame_data_location[local_lims]["mean"]
-        local_std = self.frame_data_location[local_lims]["std"]
+        data_img_input = None
+        data_img_output = None
+        if self.cache_data:
+            key_tuple = (video_index, img_index)
+            if key_tuple in self.data_cache:
+                data_img_input = self.data_cache[key_tuple]['input']
+                data_img_output = self.data_cache[key_tuple]['output']
+
+        if data_img_input is None:
+            motion_path = self.frame_data_location[video_index]["path"]
+            if not os.path.isfile(motion_path):
+                motion_path = os.path.join(
+                                 os.environ['TMPDIR'],
+                                 self.frame_data_location[video_index]['path'])
+            if not os.path.isfile(motion_path):
+                msg = 'could not find valid file path for \n'
+                msg += f"{self.frame_data_location[video_index]['path']}\n"
+                msg += f"tried\n{motion_path}\n"
+                raise RuntimeError(msg)
+
+            with h5py.File(motion_path, "r") as movie_obj:
+                data_img_input = movie_obj["data"][input_index, :, :]
+                data_img_output = movie_obj["data"][output_frame, :, :]
+
+            if self.cache_data:
+                self.data_cache[key_tuple] = {'input': data_img_input,
+                                              'output': data_img_output}
+
+        local_frame_data = self.frame_data_location[video_index]
+        local_mean = local_frame_data["mean"]
+        local_std = local_frame_data["std"]
 
-        return local_mean, local_std
+        input_full = np.zeros(
+            [1, 512, 512, len(input_index)])
+        output_full = np.zeros([1, 512, 512, 1])
+
+        data_img_input = np.moveaxis(data_img_input, 0, -1)
+
+        img_in_shape = data_img_input.shape
+        img_out_shape = data_img_output.shape
+
+        data_img_input = (data_img_input.astype(
+            "float") - local_mean) / local_std
+        data_img_output = (data_img_output.astype(
+            "float") - local_mean) / local_std
+        input_full[0, :img_in_shape[0], :img_in_shape[1]] = data_img_input
+        output_full[
+            0, :img_out_shape[0], :img_out_shape[1], 0] = data_img_output
+
+        return input_full, output_full
 
     def __data_generation__(self, index_frame):
```

Comment:
I know this isn't code you're actively changing but while you were in this file (and the others, if you could add type hints and return types to the methods that would make the code a lot cleaner for the future.